src/xdist/workermanage.py

```diff
@@ -236,10 +238,14 @@ def shutting_down(self):
     def setup(self):
         self.log("setting up worker session")
         spec = self.gateway.spec
-        args = self.config.args
+        if hasattr(self.config, "invocation_params"):
+            args = [str(x) for x in self.config.invocation_params.args or ()]
```

Comment:
Wouldn't this give unwanted arguments to the worker?

I'm investigating why pytest-cov's test suite is broken on 0.30 and invocation_params.args has lots of stuff while config.args is just the test file.

Reply:
@nicoddemus 
Can you comment, please?
(I'm currently looking into getting pytest-cov's CI to pass again so this would help there)

Reply:
Sorry @ionelmc and @blueyed, I must have missed this.

Why do you say it would give unwanted arguments to the worker? Using `invocation_params` it will initialize exactly like the master process (well almost), which solved a problem with initialization of the config file.

Reply:
@nicoddemus I guess I'm just confused about what's wrong with the latest pytest/xdist/cov combination (it's broken in some situations). Any explanation about how the all works would help me debug it.

Reply:
Not sure myself, but the latest implementation just forwards all the parameters given in the command line to the workers, making them initialize the same way as the master node.

Reply:
@nicoddemus the way I understand this the problem is that option_dict is empty in the new "invocation_params mode", thus plugins like pytest-cov don't have the necessary options to initialize. Am I missing something?

Reply:
> Thus plugins like pytest-cov don't have the necessary options to initialize.

Unless pytest-cov is trying to get the options from pytest-xdist structures instead of `config.getoption`, it should just work. The workers should now be initializing the same way as the master node/without xdist, so all the options should be there.

Reply:
@nicoddemus the problem is that none of the plugin options are there when stuff like `--dist=load --tx=popen//chdir=1` is used. When using `-n 1` everything is fine.